ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadPartitions.java

```diff
@@ -262,50 +264,48 @@ private Path locationOnReplicaWarehouse(Table table, AddPartitionDesc.OnePartiti
+    Map<String, String> lastReplicatedPartSpec = null;
+    if (!encounteredTheLastReplicatedPartition) {
+      lastReplicatedPartSpec = lastPartitionReplicated.getPartition(0).getPartSpec();
+      LOG.info("Start processing from partition info spec : {}",
+          StringUtils.mapToString(lastReplicatedPartSpec));
+    }
+
     ReplicationSpec replicationSpec = event.replicationSpec();
-    LOG.debug("table partitioned");
+    Iterator<AddPartitionDesc> partitionIterator = event.partitionDescriptions(tableDesc).iterator();
+    while (!encounteredTheLastReplicatedPartition && partitionIterator.hasNext()) {
+      AddPartitionDesc addPartitionDesc = partitionIterator.next();
+      Map<String, String> currentSpec = addPartitionDesc.getPartition(0).getPartSpec();
+      encounteredTheLastReplicatedPartition = lastReplicatedPartSpec.equals(currentSpec);
+    }
 
-    Iterator<AddPartitionDesc> iterator = event.partitionDescriptions(tableDesc).iterator();
-    while (iterator.hasNext()) {
-      /*
-      encounteredTheLastReplicatedPartition will be set, when we break creation of partition tasks
-      for a table, as we have reached the limit of number of tasks we should create for execution.
-      in this case on the next run we have to iterate over the partitions desc to reach the last replicated
-      partition so that we can start replicating partitions after that.
-       */
-      AddPartitionDesc addPartitionDesc = iterator.next();
-      if (encounteredTheLastReplicatedPartition && tracker.canAddMoreTasks()) {
-        Map<String, String> partSpec = addPartitionDesc.getPartition(0).getPartSpec();
-        Partition ptn;
-
-        if ((ptn = context.hiveDb.getPartition(table, partSpec, false)) == null) {
-          if (!replicationSpec.isMetadataOnly()) {
-            forNewTable();
-          }
-        } else {
-          // If replicating, then the partition already existing means we need to replace, maybe, if
-          // the destination ptn's repl.last.id is older than the replacement's.
-          if (replicationSpec.allowReplacementInto(ptn.getParameters())) {
-            if (replicationSpec.isMetadataOnly()) {
-              tracker.addTask(alterSinglePartition(addPartitionDesc, replicationSpec, ptn));
-              if (iterator.hasNext() && !tracker.canAddMoreTasks()) {
-                tracker.setReplicationState(
-                    new ReplicationState(new PartitionState(table.getTableName(), addPartitionDesc)
-                    )
-                );
-              }
-            } else {
-              forNewTable();
+    while (partitionIterator.hasNext() && tracker.canAddMoreTasks()) {
+      AddPartitionDesc addPartitionDesc = partitionIterator.next();
+      Map<String, String> partSpec = addPartitionDesc.getPartition(0).getPartSpec();
+      Partition ptn = context.hiveDb.getPartition(table, partSpec, false);
+      if (ptn == null) {
+        if (!replicationSpec.isMetadataOnly()) {
```

Comment:
metadataOnly flag is always false in case of bootstrap. So, shall avoid checking it here.

Reply:
for view it will still be isMetadataOnly though it should not come to partitions load for it, i looked at the code and looks like we can remove a larger part of the code now with this condition not true, will create a bug for it as it will require going through some code branches to figure this out.